trees/trees.go

```diff
@@ -152,7 +153,12 @@ func Signer(ctx context.Context, sf keys.SignerFactory, tree *trillian.Tree) (*t
 		return nil, err
 	}
 
-	signer, err := sf.NewSigner(ctx, tree.PrivateKey)
+	var keyProto ptypes.DynamicAny
+	if err := ptypes.UnmarshalAny(tree.PrivateKey, &keyProto); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal tree.PrivateKey: %v", err)
+	}
+
+	signer, err := sf.NewSigner(ctx, keyProto.Message)
```

Comment:
Is NewSigner resilient against a nil keyProto.Message? Should we test for that (here or at NewSigner)? Otherwise an incomplete request might cause a panic down the line.